route/src/route_generator_worker.cpp

```diff
@@ -460,16 +470,48 @@ namespace route {
                 cte_count_++;
 
                 if(cte_count_ > cte_count_max_) //If the distance exceeds the crosstrack distance a certain number of times, report that the route has been departed
-                    return true;
+                    {
+                        cte_count_ = 0;
+                        return true;
+                    }
+                else 
+                    return false;
+            }
+        else
+            {
+                cte_count_ = 0;
+                return false;
             }
  
     }
 
 
-    void RouteGeneratorWorker::set_CTE_counter(int cte_counter)
+
+    lanelet::ConstLanelet RouteGeneratorWorker::get_closest_lanelet_from_vector_llts (std::vector<std::pair<double, lanelet::ConstLanelet::ConstType>> list_of_pair)
     {
-        cte_count_ = cte_counter;
+        std::vector<std::pair<double, lanelet::ConstLanelet::ConstType>>  filtered;
+        double min = 1000.0;
+        lanelet::ConstLanelet min_llt;
+        for(auto i : list_of_pair)
+        {
+            for(auto j: route_llts)
+            {
+                if(i.second.id() == j.id())
+                    filtered.push_back(i);
+            }
+
+        }
+
+        for (auto i: filtered)
+        {
+            if(i.first < min)
+                {
+                    min = i.first;
+                    min_llt = i.second;
+                }
+        }
```

Comment:
if filtered is empty, we are returning empty lanelet. So if filtered is empty, we should loop through route_llts and record each distance to the point. And return the one with minimum distance. I think current_position (name not verbatim) should be a member variable.

Reply:
Filtered would only be empty if route_llt is empty though, right? Or maybe I'm missing a detail that might mean we're not getting all the lanelets?

Reply:
Filtered might be empty if the closest lanelet pairs to the point in the input, for some reason, does not contain any route lanelets. Chances are low.